library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.util.DisplayMetrics
+import androidx.annotation.MainThread
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.internal.AsyncImageLoadingCallback
+import com.datadog.android.sessionreplay.internal.utils.Base64Utils
+import com.datadog.android.sessionreplay.internal.utils.DrawableUtils
+import com.datadog.android.sessionreplay.model.MobileSegment
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.RejectedExecutionException
+
+@Suppress("UndocumentedPublicClass")
+class Base64Serializer internal constructor(
+    private var threadPoolExecutor: ExecutorService,
+    private var drawableUtils: DrawableUtils = DrawableUtils(),
+    private var base64Utils: Base64Utils = Base64Utils
+) {
+
+    private lateinit var asyncImageLoadingCallback: AsyncImageLoadingCallback
+
+    // region internal
+
+    @MainThread
+    internal fun handleBitmap(
+        displayMetrics: DisplayMetrics,
+        drawable: Drawable,
+        imageWireframe: MobileSegment.Wireframe.ImageWireframe
+    ) {
+        asyncImageLoadingCallback.startLoadingImage()
+
+        val bitmap = drawableUtils.createBitmapFromDrawable(drawable, displayMetrics)
+
+        if (bitmap == null) {
+            asyncImageLoadingCallback.finishLoadingImage()
+            return
+        }
+
+        serialiseBitmap(bitmap, imageWireframe, asyncImageLoadingCallback)
+    }
+
+    internal fun registerAsyncLoadingCallback(
+        asyncImageLoadingCallback: AsyncImageLoadingCallback
+    ) {
+        this.asyncImageLoadingCallback = asyncImageLoadingCallback
+    }
+
+    // endregion
+
+    // region testing
+
+    @VisibleForTesting
+    internal fun getThreadPoolExecutor(): ExecutorService = threadPoolExecutor
```

Comment:
I added it for testing the factory, because otherwise the factory tests can't access and validate that the executors are the same for multiple builds

Reply:
This was added for the factory tests. Since the factory has to be a singleton, I can't inject the base64serializer and therefore can't capture the constructor arguments. So I'm using this method to ensure that the same threadpoolexecutor is being passed to the different instances

Reply:
I think we'll still need this to test that the `ThreadPoolExecutor` in the builder is the same for different instances